### ISSUE TITLE
Research: use micro gateway instead of self generated web handler

### ISSUE THREAD

**Issue by kulmann** [owncloud/ocis-settings#6]
## Current Situation
For our http json endpoints (to be used by clients outside of microservices):
- we define endpoints through annotation in [protobuf file (click for example)](https://github.com/owncloud/ocis-settings/blob/56bbab28e77f957cda6bd15245fccbe728be6dc9/pkg/proto/v0/settings.proto#L36)
- from that we generate 
    a) a swagger file and a javascript client
    b) a `.pb.web.go` file with handlers for forwarding the requests into the correct service handler

Issues at the moment: we are not able to define REST JSON endpoints, because the code generation in b) lacks parsing and mapping GET query parameters into the appropriate protobuf message fields. To mitigate this, we decided to define POST requests only, since we only use the generated javascript client anyway.

This might cause problems with proper GET request caching etc in the future. Discussing this with @butonic we came up with the following idea.

## Idea
Go micro has an [API gateway](https://micro.mu/docs/api.html). That should be capable of mapping endpoints into methods of registered services, as long as we stick to their [pattern for endpoint structure](https://micro.mu/docs/api.html#api). It looks like this is already doing what we are trying to do with the custom generated .pb.web.go files. If we can use this, we can get rid of the generated .pb.web.go files, while still use our swagger definition based generated javascript client.

**Comment by kulmann** [comment]
@refs we have to talk about this :-)

**Comment by refs** [comment]
yeah, this was the other approach. We can stick to using Micro's api, which is equivalent to what we do with `POST` requests only to our proxy and forwarding it to the gRPC service.

**Comment by refs** [comment]
@kulmann we can pair tomorrow to create a PoC

**Comment by kulmann** [comment]
We want to check if the API gateway does what we need. The settings service already lives in the api namespace, so we just have to change the proxy config to route settings requests through the api gateway instead of directly to the settings service. And of course change the protobuf endpoint annotations for swagger extraction.

Related links:
https://github.com/micro/go-micro/blob/master/api/handler/api/util_test.go
https://github.com/micro/go-micro/blob/5efb3862243bda7b66ccfa293cbd0a8ec93899a0/api/handler/api/util.go#L21

**Comment by kulmann** [comment]
In an ideal world
- we would be able to use proper request methods (at least GET, POST, PUT, DELETE and PATCH)
- the api endpoints would not be hardcoded in the ocis-proxy config
- the statically served web apps (= extensions for ocis-web) could be registered or discovered (again: without hardcoded ocis-proxy config)
